app/controllers/newsletter_subscriptions_controller.rb

```diff
@@ -14,6 +20,25 @@ def subscribe
     end
   end
 
+  def confirmed
+    return redirect_to root_path, alert: 'Neplatný odkaz na potvrdenie newslettra' unless params[:token].present?
+
+    begin
+      token_data = Rails.application.message_verifier(:newsletter_confirmation).verify(params[:token])
+
+      if token_data.is_a?(Hash) && token_data[:expires_at]
+        if Time.current.to_i > token_data[:expires_at]
+          redirect_to root_path, alert: 'Platnosť odkazu na potvrdenie newslettra vypršala'
+          return
+        end
+      end
+
+      render :confirmed
+    rescue ActiveSupport::MessageVerifier::InvalidSignature
+      redirect_to root_path, alert: 'Neplatný odkaz na potvrdenie newslettra'
+    end
+  end
+
```

Comment:
@tomasdrga nevie toto za nás robiť Brevo? Že v SubscribeToNewsletterJob implementujeme to Brevo, kde dáme sbscribe emailu do nejakého listu. A tiež pošleme, že to má byť DOI (double opt in). A potom už Brevo pošle nejaký verifikačný email a vyrieši tú logiku potvrdzovania.